### ISSUE TITLE
Making PR artifacts available via GitHub Pages?

### ISSUE THREAD

**Issue by ronaldtse** [actions-mn/build-and-publish#4]
From @hasel001 of @iho-ohi :

> Right now, to get to the Word document, I have been using the process below:
> 
> a. Go to the Pull request in question
> b. Go to Checks
> c. If it exists, download the zip file from Artifacts
> c2. If it has expired, re-run the processes to regenerate the artifacts.
> d. Unzip/expand to get to the file.
> 
> Is there a process that can more easily make the documents available (especially to more novice GitHub users)?

It looks possible to deploy PR builds to a GitHub Pages site:
* https://github.com/marketplace/actions/deploy-pr-preview
 
However we will need to deal with retentions (i.e. once the PR is merged, we want to delete those files?).

Can we integrate it with some action-mn we have?

**Comment by CAMOBAP** [comment]
I think this can be easily extended to support `deployment-pr-preview` by adding one more step at the bottom build job

```yml
name: generate

on:
  workflow_call:
  pull_request:
  push:
    branches: [main]

permissions:
  contents: read
  pages: write
  id-token: write

concurrency:
  group: "pages"
  cancel-in-progress: true

jobs:
  build:
    runs-on: ubuntu-latest
    container:
      image: metanorma/metanorma:latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Cache Metanorma assets
        uses: actions-mn/cache@v1

      - name: Metanorma generate site
        uses: actions-mn/build-and-publish@v1
        with:
          agree-to-terms: true

      - name: Deploy preview
        if: ${{ github.event_name== 'pull_request' }}
        uses: rossjrw/pr-preview-action@v1
        with:
          source-dir: _site
          preview-branch: gh-pages
          umbrella-dir: pr-preview
          action: auto

  deploy:
    if: ${{ github.ref == 'refs/heads/main' }}
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1
```

@ronaldtse I think it's hard to implement this at the actions level but we can update our reusable workflow  https://github.com/metanorma/ci/blob/main/.github/workflows/user-docker.yml

Also to prevent overriding PR preview we may need to move out of `actions/deploy-pages@v1` to `JamesIves/github-pages-deploy-action@v4`

@ronaldtse maybe you can suggest some repository to try it out? Also should we implement this for our mn-samples-* repos too?

**Comment by ronaldtse** [comment]
@CAMOBAP the question here is what happens when we have multiple PRs at once?

Does the `preview-branch: gh-pages` parameter have to change for every branch? e.g.
```yaml
preview-branch: ${{github.event.pull_request.number}}
```

**Comment by CAMOBAP** [comment]
> the question here is what happens when we have multiple PRs at once?

Looks like https://github.com/rossjrw/pr-preview-action/blob/main/action.yml#L30-L31 is smart enough to separate different pull_request sub-events (actions) i.e. `deploy`, `remove`, `auto` (will be calculated It).

Also, it deploys different PR to different sub-directories [`pr-preview/pr-${PR_NUMBER}`](https://github.com/rossjrw/pr-preview-action/blob/main/action.yml#L65) so it will be able to manager each PR content separately, and delete it on actions=close

> Does the preview-branch: gh-pages parameter have to change for every branch? 

Nope we should keep it `gh-pages` unless we have our own requirements to change it. `rossjrw/pr-preview-action@v1` doesn't require to change it

So it's pretty straight forward to use it

**Comment by ronaldtse** [comment]
That's excellent, let's try this out on the mn-samples-iho repository.

**Comment by CAMOBAP** [comment]
Quick Summary  there are several issues discovered during integration of https://github.com/marketplace/actions/deploy-pr-preview:
 - GH Pages return 404 for deployed pages
 - Sometimes deployment doesn't happen during permissions issues but acting stays green

Investigation/experiments will be continues on https://github.com/metanorma/mn-samples-iho/pull/80 PR

**Comment by CAMOBAP** [comment]
It works now (with no changes since last attempt) https://github.com/metanorma/mn-samples-iho/pull/83, now I have to add it to build-and-publish strep somehow or think about another approach

**Comment by CAMOBAP** [comment]
New action implemented https://github.com/actions-mn/deploy-pages and tested on mn-samples-iho

@ronaldtse should we apply it to all our document repos

**Comment by ronaldtse** [comment]
> New action implemented https://github.com/actions-mn/deploy-pages and tested on mn-samples-iho
> 
> @ronaldtse should we apply it to all our document repos

Yes we definitely should do that! Can you create a new issue for this? Thanks!
